internal/pkg/template/override/override_test.go

```diff
@@ -4,29 +4,333 @@
+		next:      end,
+	}
+	head := &ruleNode{
+		mapVal:    mapNode{key: "Resources"},
+		valueType: mapType,
+		next:      taskDefNode,
+	}
+	return head
+}
+
+func addRequiresCompatibilities() *ruleNode {
+	var node yaml.Node
+	_ = yaml.Unmarshal([]byte(`RequiresCompatibilities[-]: EC2`), &node)
+
+	node1 := &ruleNode{
+		valueType: seqType,
+		seqVal: seqNode{
+			key:          "RequiresCompatibilities",
+			appendToLast: true,
+		},
+		endVal: node.Content[0].Content[1],
+	}
+	return newTaskDefPropertyNode(node1)
+}
+
+func addLinuxParametersCapabilities() *ruleNode {
+	var node yaml.Node
+	_ = yaml.Unmarshal([]byte(`ContainerDefinitions[0].LinuxParameters.Capabilities.Add: ["AUDIT_CONTROL", "AUDIT_WRITE"]`), &node)
+
+	node2 := &ruleNode{
+		mapVal:    mapNode{key: "Add"},
+		valueType: mapType,
+		endVal:    node.Content[0].Content[1],
+	}
+	node1 := &ruleNode{
+		mapVal:    mapNode{key: "Capabilities"},
+		valueType: mapType,
+		next:      node2,
+	}
+	return newLinuxParameters(node1)
+}
+
+func addLinuxParametersCapabilitiesInitProcessEnabled() *ruleNode {
+	node2 := &ruleNode{
+		mapVal:    mapNode{key: "InitProcessEnabled"},
+		valueType: mapType,
+		endVal: &yaml.Node{
+			Kind:  8,
+			Tag:   nodeTagBool,
+			Value: "true",
+		},
+	}
+	node1 := &ruleNode{
+		mapVal:    mapNode{key: "Capabilities"},
+		valueType: mapType,
+		next:      node2,
+	}
+	return newLinuxParameters(node1)
+}
+
+func newLinuxParameters(nextNode *ruleNode) *ruleNode {
+	node2 := &ruleNode{
+		mapVal:    mapNode{key: "LinuxParameters"},
+		valueType: mapType,
+		next:      nextNode,
+	}
+	node1 := &ruleNode{
+		valueType: seqType,
+		seqVal: seqNode{
+			key:   "ContainerDefinitions",
+			index: 0,
+		},
+		next: node2,
+	}
+	return newTaskDefPropertyNode(node1)
+}
+
+func addUlimits() *ruleNode {
+	var node yaml.Node
+	_ = yaml.Unmarshal([]byte("ContainerDefinitions[0].Ulimits[-].HardLimit: !Ref ParamName"), &node)
+
+	node3 := &ruleNode{
+		mapVal:    mapNode{key: "HardLimit"},
+		valueType: mapType,
+		endVal:    node.Content[0].Content[1],
+	}
+	node2 := &ruleNode{
+		seqVal: seqNode{
+			key:          "Ulimits",
+			appendToLast: true,
+		},
+		valueType: seqType,
+		next:      node3,
+	}
+	node1 := &ruleNode{
+		valueType: seqType,
+		seqVal: seqNode{
+			key:   "ContainerDefinitions",
+			index: 0,
+		},
+		next: node2,
+	}
+	return newTaskDefPropertyNode(node1)
+}
+
+func exposeExtraPort() *ruleNode {
+	node3 := &ruleNode{
+		mapVal:    mapNode{key: "ContainerPort"},
+		valueType: mapType,
+		endVal: &yaml.Node{
+			Kind:  8,
+			Tag:   nodeTagInt,
+			Value: "5000",
+		},
+	}
+	node2 := &ruleNode{
+		valueType: seqType,
+		seqVal: seqNode{
+			key:          "PortMappings",
+			appendToLast: true,
+		},
+		next: node3,
+	}
+	node1 := &ruleNode{
+		valueType: seqType,
+		seqVal: seqNode{
+			key:   "ContainerDefinitions",
+			index: 0,
+		},
+		next: node2,
+	}
+	return newTaskDefPropertyNode(node1)
+}
+
+func referBadSeqIndex() *ruleNode {
```

Comment:
Eventually we'll be able to remove all these helpers right? and instead test only the public functions with a template and list of rules

Reply:
yes!